### DISCUSSION TITLE
Is the OTP expiration/length configurable within self hosted Supabase?

### DISCUSSION THREAD

**Discussion by zieddhf** [supabase/supabase#26330] [Questions]
Lately we noticed some warnings within Supabase advisor suggesting to set the OTP expiry less than an hour and another warning about the OTP length.
After checking GoTrue docs i found these variables :
```
MAILER_OTP_EXP - number
Controls the duration an email link or otp is valid for.

SMS_OTP_EXP - number
Controls the duration an sms otp is valid for.

SMS_OTP_LENGTH - number
Controls the number of digits of the sms otp sent.
```
i added the above variables within GoTrue docker service environment.
inspecting GoTrue running container environment variables to check if the configuration is applied results : 
`GOTRUE_MAILER_OTP_EXP=3600, GOTRUE_SMS_OTP_LENGTH=8`

The problem is Supabase dashboard still showing the default OTP configuration (EXP=86400, LENGTH=6), alongside the warnings.

My question is it possible to tweak OTP configuration within docker self hosted Supabase? if so i will appreciate any guidance to figure out how.

**Comment by tecoad** [comment]
+1

**Comment by riccardolardi** [comment]
Any update on this?

**Comment by phurikornpt** [comment]
i try to config `GOTRUE_MAILER_OTP_EXP : 30`   number is sec it work for me.
ref =>  https://github.com/supabase/cli/blob/develop/internal/start/start.go#L468

**Comment by snw7** [comment]
for sms `GOTRUE_SMS_OTP_EXP: ${SMS_OTP_EXPIRY}` works too in the self hosted docker setup! 😄
